Scrubber Clog Random Event Spiritual Successor -- New janitor content!

Rhials opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

![Janitorial Content](https://user-images.githubusercontent.com/28870487/168494700-38b6f823-115d-4b87-975a-2ad20fe19225.png)

Adds a brand new random event as a spiritual successor to the long-lost scrubber overflow event! Scrubbers (that is, a single one on the station) may now randomly become clogged with mobs, and after an alert, will begin pumping out mobs at a steady rate. The infestation can be fixed by either welding the scrubber shut and waiting for the infestation to clear, or you can use a plunger on the scrubber to clear it manually.

This event comes in four flavors:

Minor scrubber clogs will spawn harmless nuisance mobs, and is very common. A brief diversion that gets the janitor moving about the place and talking to people. 

Major scrubber clogs will spawn more hostile mobs, and is much less common. Nothing super dangerous, but enough to get people screaming for the janitor over the radio.

Critical scrubber clogs have a unique alert, and will infest a scrubber with some particularly dangerous mobs. Probably not a good idea to tackle this as a lone janitor.

"Strange" scrubber clogs are very rare, and is meant to spawn mobs I deemed too silly for the other events. The mental image of an entire bear crawling out of the ventilation ducts is funny, but would get old very fast.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

Janitor content meant to break up the monotony of mopping up the corpses of your recently-flayed coworkers. The rarer event subtypes are also a good way of injecting some peril/silliness into a round. Also, when's the last time you've seen a janitor using their plunger? 

Fikou said he wants more random events, too. That's worth something, right?

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
add: Adds a new random event -- Scrubber Clog! Scrubbers may now randomly be clogged by mobs, which will result in them crawling out and getting all over the place. This can be fixed by pumping the scrubber with a plunger, or by just welding the scrubber shut and leaving it for a while. 
add: The plunger has been integrated into the janitor's arsenal. You may now purchase them from the Janidrobe and equip them into janibelts.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Rhials committed:
scrubber clog event basics

very very wip please dont laugh at me

Rhials committed:
event actually works now whoopee

dentistry

Rhials committed:
modifies vars

Rhials committed:
Severity levels, mob spawn lists

Rhials committed:
what the dog doin

Rhials committed:
signals made me pee my pants

Rhials committed:
OOPS

Rhials committed:
clog/unclog basic functionality

:)

Rhials committed:
spacemandmm doesnt like sleeps

Rhials committed:
Minor tweaks, mob spawning doesn't work, fix on the way

Rhials committed:
Mob spawning now works, and the event should now work with more than 1 scrubber on station

also a new message for critical overflows

Rhials committed:
cleaner, leaner code

Rhials committed:
changes file name to scrubber_clog because I dont like vent_clog

Rhials committed:
A lot of stuff

Tweaks on max spawns, spawn delays, to be roughly scaled against severity (more dangerous mobs means slower spawn rate and lower max).

New "Strange" severity with sillier or more useful mob list.

General tweaks to the mob lists, made with the max/min spawns in mind.

Rhials committed:
more tweaks

Rhials committed:
jesus fucking christ

refactors a lot of the actual meat of the mob spawning. Complete shitfest, why did I even do this? Code is a bit leaner now I guess???

Rhials committed:
fixes event timings, removes debug alerts

also adds groundwork for getting the list working

Rhials committed:
oops

Rhials committed:
everything works now I think

Rhials committed:
oops

Rhials committed:
final event length tweaks. Also a typo lol

Rhials committed:
fixes potential unclogging bug

Rhials committed:
Update scrubber_clog.dm

Rhials committed:
forgot to add this

MrMelbert commented on code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm:
```diff
@@ -463,5 +468,35 @@
+		return
+
+	to_chat(user, span_notice("You begin pumping the [name] with your plunger."))
+	if(do_after(user, 60, target = src))
```
```suggestion
	if(do_after(user, 6 SECONDS, target = src))
```

MrMelbert commented on code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm:
```diff
@@ -463,5 +468,35 @@
+	clogged = TRUE
+
+/obj/machinery/atmospherics/components/unary/vent_scrubber/proc/produce_mob(datum/source, spawned_mob, living_mobs)
+	if(welded == FALSE)
```
```suggestion
	if(welded)
		return
```
1. you don't need to check for `== FALSE`, and 
2. you can early return this

MrMelbert commented on code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm:
```diff
@@ -463,5 +468,35 @@
+
+/obj/machinery/atmospherics/components/unary/vent_scrubber/proc/produce_mob(datum/source, spawned_mob, living_mobs)
+	if(welded == FALSE)
+		var/mob/new_mob = new spawned_mob(src.loc)
```
```suggestion
		var/mob/new_mob = new spawned_mob(loc)
```

MrMelbert commented on code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm:
```diff
@@ -463,5 +468,35 @@
+	if(welded == FALSE)
+		var/mob/new_mob = new spawned_mob(src.loc)
+		living_mobs += new_mob
+		src.visible_message(span_warning("[new_mob] crawls out of [name]!"))
```
```suggestion
		visible_message(span_warning("[new_mob] crawls out of [src]!"))
```

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,144 @@
+/datum/round_event/scrubber_clog
+	announceWhen = 1
+	startWhen = 10
+	endWhen = 6000
```
All our events are very old and don't use the seconds / minutes defines, you should convert these over. 
```suggestion
	endWhen = 10 MINUTES
```

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,144 @@
+	announceWhen = 1
+	startWhen = 10
+	endWhen = 6000
+	var/obj/machinery/atmospherics/components/unary/vent_scrubber/scrubber //Scrubber selected for the event
+	var/mob/spawned_mob = /mob/living/basic/cockroach //What mob will be spawned
+	var/severity = "Minor" //Severity of the event (how dangerous are the spawned mobs, and at what quantity)
+	var/maximum_spawns = 3 //Cap on the number of spawned mobs that can be alive at once
+	var/spawn_delay = 10 //Interval between mob spawns
+	var/list/living_mobs = list() //Used to track/limit produced mobs
+	var/clogged = TRUE //Used for tracking if the clog signal should be sent
```
Convert these to auto-doc comments. 
Use three slashes `///`, and put them the line above.

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,144 @@
+	announceWhen = 1
+	startWhen = 10
+	endWhen = 6000
+	var/obj/machinery/atmospherics/components/unary/vent_scrubber/scrubber //Scrubber selected for the event
```
I believe this reference needs to be handled.
You can use a weakref or a signal, and then handle cleanup on event end.

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,144 @@
+	endWhen = 6000
+	var/obj/machinery/atmospherics/components/unary/vent_scrubber/scrubber //Scrubber selected for the event
+	var/mob/spawned_mob = /mob/living/basic/cockroach //What mob will be spawned
+	var/severity = "Minor" //Severity of the event (how dangerous are the spawned mobs, and at what quantity)
```
These severities should be defines. 
`#define minor_severity_clog "Minor"`

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,144 @@
+	if(!scrubber)
+		CRASH("Unable to find suitable scrubber.")
+	spawned_mob = get_mob()
+	endWhen = rand(5000,7000)
+	maximum_spawns = rand(3, 5)
+	spawn_delay = rand(10,15)
```
Seconds/minute defines

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,144 @@
+/datum/round_event/scrubber_clog/proc/get_mob()
+	switch(severity)
+		if("Minor") //Spawns completely harmless nuisance mobs.
+			var/list/minor_mobs = list(
```
These all can be static lists.

MrMelbert commented on code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm:
```diff
@@ -38,9 +38,14 @@
 	COOLDOWN_DECLARE(check_turfs_cooldown)
 
 /obj/machinery/atmospherics/components/unary/vent_scrubber/New()
+	RegisterSignal(src, COMSIG_VENT_CLOG, .proc/clog)
+	RegisterSignal(src, COMSIG_VENT_UNCLOG, .proc/unclog)
```
This actually seems a bit like signal misuse. 

`clog()` can just be called directly from the event on the scrubber, and doesn't need a signal. 
`unclog()` can also be called from either the event when it's done, or a plunger.

MrMelbert commented on code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm:
```diff
@@ -463,5 +468,35 @@
+
+/obj/machinery/atmospherics/components/unary/vent_scrubber/proc/clog(datum/source)
+	SIGNAL_HANDLER
+	RegisterSignal(src, COMSIG_PRODUCE_MOB, .proc/produce_mob)
```
Same here. Instead of being a signal, you can just call "produce_mob" directly, instead of controlling it via signal.

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,144 @@
+
+/datum/round_event/scrubber_clog/strange/setup()
+	. = ..()
+	endWhen = rand(6000,8000)
```
time defines

Rhials replied:
![image](https://user-images.githubusercontent.com/28870487/168923478-2bf23828-f77d-48ef-9c16-ae2ae0761666.png)

the "activefor" timer that gets used for starting/ending/etc only counts up once a second, not in deciseconds. Using timedefines for some of the stuff in here will make it last ten times longer than intended. I'm gonna go back and undo some of the defines I added, but I just wanted to give a reason why first.

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,144 @@
+/datum/round_event/scrubber_clog/strange/setup()
+	. = ..()
+	endWhen = rand(6000,8000)
+	maximum_spawns = 3
```
These hardcoded numbers (the ones you don't use rand for) can be set in the subtype definition.

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,144 @@
+
+/datum/round_event/scrubber_clog/critical/setup()
+	. = ..()
+	maximum_spawns = 3
```
(ditto)

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,144 @@
+
+/datum/round_event/scrubber_clog/proc/life_check()
+	for(var/mob/living/mob_check in living_mobs)
+		if(mob_check.health <= 0 || !mob_check.health)
```
```suggestion
		if(mob_check.health <= 0)
```
`!health` is the same as checking `health == 0`.

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,144 @@
+	var/severity = "Minor" //Severity of the event (how dangerous are the spawned mobs, and at what quantity)
+	var/maximum_spawns = 3 //Cap on the number of spawned mobs that can be alive at once
+	var/spawn_delay = 10 //Interval between mob spawns
+	var/list/living_mobs = list() //Used to track/limit produced mobs
```
This also needs to have the references handled when the event ends. 
Reference handing is a bit complex so feel free to ping me with questions.

MrMelbert commented on code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm:
```diff
@@ -463,5 +468,35 @@
+/obj/machinery/atmospherics/components/unary/vent_scrubber/proc/unclog()
+	clogged = FALSE
+
+/obj/machinery/atmospherics/components/unary/vent_scrubber/proc/is_clogged()
+	return clogged
```
This getter's a bit superfluous, you can just check `clogged`.
Normally you'd only implement something like this if you planned on overriding behavior for subtypes.

MrMelbert commented on code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm:
```diff
@@ -463,5 +468,35 @@
+		to_chat(user, span_notice("You finish pumping the [name]."))
+		clogged = FALSE
+
+/obj/machinery/atmospherics/components/unary/vent_scrubber/proc/clog(datum/source)
```
Auto-docs for new procs! Same goes for others.

MrMelbert commented on code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm:
```diff
@@ -463,5 +468,35 @@
+
+	to_chat(user, span_notice("You begin pumping the [name] with your plunger."))
+	if(do_after(user, 60, target = src))
+		to_chat(user, span_notice("You finish pumping the [name]."))
```
```suggestion
		to_chat(user, span_notice("You finish pumping [src]."))
```
`src` will add in `the` for you.

MrMelbert commented on code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm:
```diff
@@ -463,5 +468,35 @@
+		to_chat(user, span_notice("You cannot pump the [name] if it's welded shut!."))
+		return
+
+	to_chat(user, span_notice("You begin pumping the [name] with your plunger."))
```
```suggestion
	to_chat(user, span_notice("You begin pumping [src] with your plunger."))
```

MrMelbert commented on code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm:
```diff
@@ -463,5 +468,35 @@
+		return
+
+	if(welded)
+		to_chat(user, span_notice("You cannot pump the [name] if it's welded shut!."))
```
```suggestion
		to_chat(user, span_notice("You cannot pump the [src] if it's welded shut!."))
```

MrMelbert commented on code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm:
```diff
@@ -463,5 +468,35 @@
 	piping_layer = 4
 	icon_state = "scrub_map_on-4"
 
+/obj/machinery/atmospherics/components/unary/vent_scrubber/plunger_act(obj/item/plunger/P, mob/living/user, reinforced)
```
```suggestion
/obj/machinery/atmospherics/components/unary/vent_scrubber/plunger_act(obj/item/plunger/plunger, mob/living/user, reinforced)
```
Better var names. Don't need to change all the other implementations, just this one is fine.

MrMelbert commented on code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm:
```diff
@@ -38,9 +38,14 @@
 	///Radio connection from the air alarm
 	var/radio_filter_in
 
+	//Enables the use of plunger_act for ending the vent clog random event
+	var/clogged = FALSE
```
You can probably use a trait to track this, instead of a var, but the var is fine. Just an idea.

MrMelbert reviewed (changes requested):
Preliminary review. Likely will be more following up, but this is a great start + funny event. Good job!

Rhials committed:
first fix batch, more to come

Rhials committed:
adds second/minutes defines. Also tweaks some of said values.

Rhials committed:
fixes signal chicanery, moves hardcoded stuff out of setup

Rhials committed:
OOPS

MetalClone commented:
This is really cool and props for this idea. Thanks for giving janitors something they deserve to get paid for.

Rhials committed:
i fucking ahte you dm i hate you so much i hate you i hate you i hate you i hate you

fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you fuck you

YakumoChen commented:
Smoke clogs you will be missed :(

LemonInTheDark commented:
Do janitors get a plunger? I thought that was a chemistry tool

LemonInTheDark commented:
Do janitors get a plunger? I thought that was a chemistry tool

Rhials commented:
Huh, now that you mention it, janitors don't actually get plungers. I'll fix that in a bit. Whoops.

Rhials committed:
adds plungers to the janitor's vendor. Oops.

Rhials committed:
Merge branch 'master' into newscrubberevent

Rhials committed:
weird indents

Rhials committed:
plungers now fit in janibelt. Also removes a weird indent in belts.dm.

Rhials committed:
Auto-docs for new procs!

also punctuation

Rhials committed:
moves procs, makes lists static and defined at the type level in get_mob.

Rhials committed:
undoes time defines on some stuff

Rhials committed:
Removed the now obselete severity var

Rhials committed:
i should actually build these before I commit them lol

Rhials committed:
ok

Rhials committed:
handles signals

MrMelbert commented on code/modules/atmospherics/machinery/components/unary_devices/vent_scrubber.dm:
```diff
@@ -463,5 +466,57 @@
+ * Relevant code for how the list is handled is in the scrubber_clog.dm file.
+ */
+
+/obj/machinery/atmospherics/components/unary/vent_scrubber/proc/produce_mob(spawned_mob, living_mobs)
```
```suggestion
/obj/machinery/atmospherics/components/unary/vent_scrubber/proc/produce_mob(spawned_mob, list/living_mobs)
```

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,193 @@
+	scrubber.produce_mob() //The first one's free!
+
+/datum/round_event/scrubber_clog/tick() //Checks if spawn_interval is met, then sends signal to scrubber to produce a mob.
+	if(activeFor % spawn_delay == 0 && scrubber.clogged == TRUE)
```
```suggestion
	if(activeFor % spawn_delay == 0 && scrubber.clogged)
```

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,193 @@
+				/mob/living/basic/cockroach,
+				/mob/living/simple_animal/butterfly
+				)
+	return(pick(mob_list))
```
```suggestion
	return pick(mob_list)
```

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,193 @@
+
+/datum/round_event/scrubber_clog/start() //Sets the scrubber up for unclogging/mob production.
+	scrubber.clog()
+	scrubber.produce_mob() //The first one's free!
```
You aren't passing any arguments into `produce_mob()` here, and it expects some from what I can tell?

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,193 @@
+ */
+
+/datum/round_event/scrubber_clog/proc/life_check()
+	for(var/mob/living/mob_check in living_mobs)
+		if(mob_check.health <= 0)
+			living_mobs -= mob_check
```
because `living_mobs` is full of weakrefs, you will need to resolve them. 
```suggestion
	for(var/datum/weakref/mob_ref as anything in living_mobs)
		var/mob/living/real_mob = mob_ref.resolve()
		if(QDELETED(real_mob) || real_mob.health <= 0) // Note, you can also check for real_mob.stat == DEAD, may be a bit more explicit than health less than zero
			living_mobs -= mob_ref 
```

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,193 @@
+ * Handles the scrubber ref if there are no valid scrubbers to replace it with.
+ */
+
+/datum/round_event/scrubber_clog/proc/scrubber_move()
```
```suggestion
/datum/round_event/scrubber_clog/proc/scrubber_move(datum/source)
```
Not 100% necessary, but the first argument of signals is always the source of the signal, so it's just good to keep track of (even if you don't use it)

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,193 @@
+	SIGNAL_HANDLER
+	scrubber = null //If by some great calamity, the last valid scrubber is destroyed, the ref is cleared.
+	scrubber = get_scrubber()
+	if(!scrubber)
+		CRASH("Unable to find suitable scrubber.")
```
Will the event end if this crashes?

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,193 @@
+
+	RegisterSignal(scrubber, COMSIG_PARENT_QDELETING, .proc/scrubber_move)
+
+	switch(pick(1,2))
+		if(1)
+			priority_announce("Lifesign readings have moved to a new location in the ventilation network. New location: [get_area(scrubber)]", "Lifesign Notification")
+		if(2)
+			priority_announce("Lifesign readings have moved to a new location in the ventilation network. New location is unknown at this time.", "Lifesign Notification")
```
You can just use a `prob(50)` for this instead of a switch. 

Also not necessary, but it can be condensed into one line. (`New Location: [prob(50) ? "Unknown.":"[get_area(scrubber)]"]`)

Also also, it may be more(?) consistent to use `get_area_name` instead of `get_area`. But I'm not 100% certain on that, it may need some testing.

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,193 @@
+		/mob/living/simple_animal/hostile/bee,
+		/mob/living/simple_animal/hostile/giant_spider
+		)
+	return(pick(mob_list))
```
```suggestion
	return pick(mob_list)
```

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,193 @@
+	var/static/list/mob_list = list(
+		/mob/living/simple_animal/hostile/rat,
+		/mob/living/simple_animal/hostile/bee,
+		/mob/living/simple_animal/hostile/giant_spider
+		)
```
```suggestion
		/mob/living/simple_animal/hostile/giant_spider,
	)
```

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,193 @@
+ */
+
+/datum/round_event/scrubber_clog/proc/get_mob()
+	var/static/list/mob_list = list(
+				/mob/living/simple_animal/mouse,
+				/mob/living/basic/cockroach,
+				/mob/living/simple_animal/butterfly
+				)
```
```suggestion
	var/static/list/mob_list = list(
		/mob/living/simple_animal/mouse,
		/mob/living/basic/cockroach,
		/mob/living/simple_animal/butterfly,
	)
Trailing commas are free, and lists should only indent once. (Make sure github isn't messing up intendation) 
```

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,193 @@
+		/mob/living/simple_animal/hostile/carp,
+		/mob/living/simple_animal/hostile/bee/toxin,
+		/mob/living/basic/cockroach/glockroach,
+		)
```
```suggestion
	)
```

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,193 @@
+		/mob/living/simple_animal/hostile/bee/toxin,
+		/mob/living/basic/cockroach/glockroach,
+		)
+	return(pick(mob_list))
```
```suggestion
	return pick(mob_list)
```

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,193 @@
+		/mob/living/simple_animal/hostile/bear,
+		/mob/living/simple_animal/pet/gondola,
+		/mob/living/simple_animal/hostile/mushroom,
+		/mob/living/simple_animal/hostile/lightgeist
+		)
```
```suggestion
		/mob/living/simple_animal/hostile/lightgeist,
	)
```

MrMelbert commented on code/modules/events/scrubber_clog.dm:
```diff
@@ -0,0 +1,193 @@
+		/mob/living/simple_animal/hostile/mushroom,
+		/mob/living/simple_animal/hostile/lightgeist
+		)
+	return(pick(mob_list))
```
```suggestion
	return pick(mob_list)
```

MrMelbert commented on code/modules/vending/wardrobes.dm:
```diff
@@ -362,7 +362,7 @@
 		/obj/item/watertank/janitor = 1,
 		/obj/item/storage/belt/janitor = 2,
-		)
+		/obj/item/plunger = 2)
```
```suggestion
		/obj/item/plunger = 2,
	)
```

MrMelbert reviewed (changes requested):
Really good documentation and work, Almost there!

Rhials committed:
fixes issues. All of them, I think

Rhials committed:
makes it so that scrubber moving continues production

Rhials committed:
ends event on crash

Rhials committed:
stinky extra line

Rhials committed:
wacky indent

I really need to look over all this shit before I start committing stuff
